### ISSUE TITLE
/bin/sh: 0: Illegal option -

### ISSUE THREAD

**Issue by firekcc** [carlossg/docker-maven#114]
### i use the java8 dockerfile to build across the problem
the mvn-entrypoint.sh has error

> /bin/sh: 0: Illegal option -

I hope you can take a look thanks

**Comment by carlossg** [comment]
what command do you use, what docker version, what sha of the docker image?

**Comment by bigshaqmnh** [comment]
Hello, I faced the same annoying issue on Windows 10 =(
My docker version is 2.1.0.1
My colleague on Mac launched the same docker commands without any problems.
It seems like something is wrong with running sh scripts, but I really struggle to find what goes wrong.
Thanks for help in advance!

**Comment by charleech** [comment]
The root cause is the **Windows** use `\r\n` as a end-of-line, while the ***nix** use only `\n`. Then inside the docker there will be a `^M` character at `mvn-entrypoint.sh` which leads us to the `/bin/sh: 0: Illegal option -` since the first line.

There are many ways to remove that `^M` character as the following example: -

* Upload the `mvn-entrypoint.sh` to the linux host and use the following command: -
  * `sed -i -e 's/\r//g' /path/to/mvn-entrypoint.sh`
  * Download it back to the **Windows** machine
* Use the well-known tools such as `dos2unix` to convert at the **Windows** machine.
* Use the text editor such as **Notepad++** to convert
  * Go to menu `Edit` -> `EOL Conversion` -> `Unix (LF)`

**Comment by carlossg** [comment]
so it was only when you built it but not when you pulled it?

Thanks for the detailed description

**Comment by charleech** [comment]
> so it was only when you built it but not when you pulled it?
> 
Yes. The official image published at the DockerHub works perfectly.

I'm quite sure that it's about the cloning this repository to the Windows machine which the Git client save as windows format, `\r\n`. 

During building it locally on the Windows machine has no issue, but when executing that image, e.g. `docker run -it --rm my/custom:maven /bin/bash`, this issue occurs.

After removing the `^M` from `mvn-entrypoint.sh` and rebuild, everything works like a charm.
